File: Matrices/Matrices.java
package Matrices;

/**
 * Demonstrates matrix declaration and element access.
 * Creates a 2x3 matrix and prints all elements in row-column format.
 */
public class Matrices {
    public static void main(String[] args) {
        Runtime runtime = Runtime.getRuntime();
        runtime.gc();
        long memoryBefore = runtime.totalMemory() - runtime.freeMemory();
        long startTime = System.nanoTime();
        
        // Initialize 2x3 matrix
        int[][] matrix = {
            {1, 2, 3},
            {4, 5, 6}
        };

        // Print matrix elements
        System.out.println("Result Matrix:");
        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
        
        long endTime = System.nanoTime();
        long memoryAfter = runtime.totalMemory() - runtime.freeMemory();
        long memoryUsed = memoryAfter - memoryBefore;
        
        // Performance statistics
        System.out.println("\n--- Performance Statistics ---");
        System.out.printf("Execution time: %.4f ms%n", (endTime - startTime) / 1_000_000.0);
        System.out.println("Memory usage: " + (memoryUsed > 0 ? memoryUsed : "negligible") + " bytes");
        System.out.println("\nTime Complexity:");
        System.out.println("  - Access: O(1)");
        System.out.println("  - Search: O(m*n)");
        System.out.println("  - Insertion (row/col): O(m*n)");
        System.out.println("  - Traversal: O(m*n)");
    }
}

File: Matrices/Matrices.py
import time
import sys


def main():
    """
    Demonstrates matrix declaration and element access.
    Creates a 2x3 matrix and prints all elements in row-column format.
    """
    start_time = time.time()
    
    # Initialize 2x3 matrix
    matrix = [[1, 2, 3], [4, 5, 6]]
    
    # Memory calculation
    memory_bytes = sys.getsizeof(matrix) + sum(sys.getsizeof(row) + sum(sys.getsizeof(item) for item in row) for row in matrix)

    # Print matrix elements
    print("Result Matrix:")
    for i in range(len(matrix)):
        for j in range(len(matrix[i])):
            print(matrix[i][j], end=" ")
        print()
    
    end_time = time.time()
    
    # Performance statistics
    print("\n--- Performance Statistics ---")
    print(f"Execution time: {(end_time - start_time) * 1000:.4f} ms")
    print(f"Memory usage: {memory_bytes} bytes")
    print("\nTime Complexity:")
    print("  - Access: O(1)")
    print("  - Search: O(m*n)")
    print("  - Insertion (row/col): O(m*n)")
    print("  - Traversal: O(m*n)")


if __name__ == "__main__":
    main()
